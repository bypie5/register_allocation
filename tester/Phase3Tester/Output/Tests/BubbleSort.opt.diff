--- SelfTestCases/BubbleSort.opt.out	2020-03-05 22:31:29.000000000 -0800
+++ ./Output/Tests/BubbleSort.opt.out	2020-03-09 14:01:49.000000000 -0700
@@ -1,22 +1,3 @@
-20
-7
-12
-18
-2
-11
-6
-9
-19
-5
-99999
-2
-5
-6
-7
-9
-11
-12
-18
-19
-20
-0
+--- Error (ProgramFault) ---
+require function address, found undefined value
+  Main() line 13
