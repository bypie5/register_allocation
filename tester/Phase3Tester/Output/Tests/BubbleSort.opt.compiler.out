func Main [in 0, out 0, local 0]
    $s0 = HeapAllocZ(8)
    Error()
    $s0 = 
    PrintIntS($s0)
    ret
func BBS.Start [in 0, out 0, local 0]
    $s1 = 
    $s1 = 
    PrintIntS(99999)
    $s0 = 
    $s0 = 
    ret
func BBS.Sort [in 0, out 0, local 0]
    $s2 = Sub($s0 1)
    $s0 = Sub(0 1)
    $s3 = LtS($s0 $s2)
    $s3 = 1
    $s4 = Add($s2 1)
    $s5 = LtS($s3 $s4)
    $s4 = Sub($s3 1)
    Error()
    $s6 = Lt($s4 $s6)
    Error()
    $s4 = MulS( 4)
    $s4 = Add($s4 )
    Error()
    $s6 = Lt($s3 $s6)
    Error()
    $s4 = MulS($s3 4)
    $s4 = Add($s4 )
    $s4 = LtS($s5 )
    $s4 = Sub($s3 1)
    Error()
    $s6 = Lt($s4 $s6)
    Error()
    $s5 = MulS($s4 4)
    $s5 = Add($s5 )
    Error()
    $s7 = Lt($s4 $s7)
    Error()
    $s4 = MulS( 4)
    $s4 = Add($s4 )
    Error()
    $s7 = Lt($s3 $s7)
    Error()
    $s5 = MulS($s3 4)
    $s5 = Add($s5 )
    Error()
    $s0 = Lt( $s0)
    Error()
    $s0 = MulS( 4)
    $s0 = Add($s0 )
    $s0 = 0
     = Add( 1)
     = Sub( 1)
    ret
func BBS.Print [in 0, out 0, local 0]
    $s0 = 0
    $s3 = LtS($s0 $s2)
    Error()
    $s3 = Lt($s0 $s3)
    Error()
    $s2 = MulS($s0 4)
    $s2 = Add($s2 )
    PrintIntS($s3)
     = Add( 1)
    ret
func BBS.Init [in 0, out 0, local 0]
    $s2 = 
    Error()
    $s2 = Lt(0 $s2)
    Error()
    $s0 = MulS(0 4)
    $s0 = Add($s0 )
    Error()
    $s2 = Lt(1 $s2)
    Error()
    $s0 = MulS(1 4)
    $s0 = Add($s0 )
    Error()
    $s2 = Lt(2 $s2)
    Error()
    $s0 = MulS(2 4)
    $s0 = Add($s0 )
    Error()
    $s2 = Lt(3 $s2)
    Error()
    $s0 = MulS(3 4)
    $s0 = Add($s0 )
    Error()
    $s2 = Lt(4 $s2)
    Error()
    $s0 = MulS(4 4)
    $s0 = Add($s0 )
    Error()
    $s2 = Lt(5 $s2)
    Error()
    $s0 = MulS(5 4)
    $s0 = Add($s0 )
    Error()
    $s2 = Lt(6 $s2)
    Error()
    $s0 = MulS(6 4)
    $s0 = Add($s0 )
    Error()
    $s2 = Lt(7 $s2)
    Error()
    $s0 = MulS(7 4)
    $s0 = Add($s0 )
    Error()
    $s1 = Lt(8 $s1)
    Error()
    $s0 = MulS(8 4)
    $s0 = Add($s0 )
    Error()
    $s1 = Lt(9 $s1)
    Error()
    $s0 = MulS(9 4)
    $s0 = Add($s0 )
    ret
func AllocArray [in 0, out 0, local 0]
    $s0 = MulS($s1 4)
    $s0 = Add($s0 4)
    $s2 = HeapAllocZ($s0)
    ret

